libsigner/src/session.rs

```diff
@@ -134,10 +131,10 @@ impl SignerSession for StackerDBSession {
     /// connect to the replica
     fn connect(
         &mut self,
-        host: SocketAddr,
+        host: &str,
         stackerdb_contract_id: QualifiedContractIdentifier,
     ) -> Result<(), RPCError> {
-        self.host = host;
+        self.host = host.to_owned();
```

Comment:
Since you're just storing the host parameter, this is actually a legitimate case to be specific and take an owned String as input. I missed this in my first review. My apologies for the extra round-trip. Feel free to revert the last change.